### PR TITLE
Animal hunter

### PR DESCRIPTION
**Pull request by WaveOfDandelions**

### CODE REVIEW

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,45 @@
...
+    if (!obj.hidden && obj instanceof Herbivore) {
+      obj.health -= 50;
+    }
+
+    Animal.alive.map((animal, index) => {
+      if(animal.health <= 0) {
+          Animal.alive.splice(index, 1);
+        }
+      return animal;
+    })
```
Comment (choeqq) [inline src/herbivoresAndCarnivores.js:41]: filter method is perfect for that

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,45 @@
...
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+  }
+
+  bite(obj) {
```
Comment (choeqq) [inline src/herbivoresAndCarnivores.js:31]: I think you can do little better on that obj name

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
```
Comment (choeqq) [inline src/herbivoresAndCarnivores.js:9]: please consider adding reduceHealth and die methods to Animal class, then you could just call them when biten

File: src/herbivoresAndCarnivores.js [hunk 4]
```diff
@@ -1,15 +1,45 @@
...
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+    this.hidden = false;
+  }
```
Comment (choeqq) [inline src/herbivoresAndCarnivores.js:18]: Herbivore should also have health
```suggestion
  constructor(name, health) {
    super(name, health);
    this.name = name;
    this.hidden = false;
  }
```

**Review by choeqq — CHANGES_REQUESTED** [review]

**Review by asakevych — APPROVED** [review]
Great job!! 🚀 🔥 👏 

you can also remove comments `// write your code here` to make file look better, there is already your code 😉

**Pull request closed without merging**